esrally/client.py

```diff
@@ -129,7 +129,17 @@ def _is_set(self, client_opts, k):
 
     def create(self):
         import elasticsearch
-        return elasticsearch.Elasticsearch(hosts=self.hosts, ssl_context=self.ssl_context, **self.client_options)
+        instance = elasticsearch.Elasticsearch(hosts=self.hosts, ssl_context=self.ssl_context, **self.client_options)
+        if self._is_set(self.client_options, "use_api_key"):
+            # an api_key is generated per client, which happens after an initial handshake using an existing auth scheme
+            # once this key is generated, a new client is created using the api_key client option and http_auth is removed
+            # the reason the options were not removed in place was because of the logic to coalesce the key tuple into a
+            # header that the client could use.
+            api_key_response = instance.security.create_api_key({"name": "rally-api-key"})
+            self.client_options.pop("http_auth")
```

Comment:
I don't know if it possible to use API keys without prior basic auth but `pop` raises a `KeyError` when the key is not present, so we should maybe do `self.client_options.pop("http_auth", None)` here, see [docs](https://docs.python.org/3/library/stdtypes.html#dict.pop).